store/sqlstore/channel_store.go

```diff
@@ -3721,10 +3712,47 @@ func (s SqlChannelStore) UpdateSidebarCategories(userId, teamId string, categori
 
 		}
 
+		// Update the favorites preferences based on channels moving into or out of the Favorites category for compatibility
 		if category.Type == model.SidebarCategoryFavorites {
+			// Remove any old favorites
+			sql, args, _ = s.getQueryBuilder().Delete("Preferences").Where(
+				sq.Eq{
+					"Name":     originalCategory.Channels,
+					"Category": model.PREFERENCE_CATEGORY_FAVORITE_CHANNEL,
+				},
+			).ToSql()
+
+			if _, err = transaction.Exec(sql, args...); err != nil {
+				return nil, model.NewAppError("SqlPostStore.UpdateSidebarCategory", "store.sql_channel.sidebar_categories.app_error", nil, err.Error(), http.StatusInternalServerError)
+			}
```

Comment:
Could we perhaps refactor this deletion code from the if and the else conditions into a single one ? Only the `Name` column needs to change. And then add the new ones if type is `SidebarCategoryFavorites`.

Reply:
I don't think there's much we could do to make that shorter since `Name` pulls from the two different arrays there. The best we could do is to make a helper function to delete multiple preferences by category and name, but I don't think that would do a ton to make it more readable.

Also, I do kind of like having two clear paths for Favorites and non-Favorites categories since the steps for updating the preferences in each case are perfectly clear with this.